### PR TITLE
v2019-04-30 GA API

### PR DESCRIPTION
**Pull request by mjudeikis**
```release-note
Add GA API v2019-04-30

Move converters inside api sub-packages and rename them to be more informative. 
v20190430.ConvertFromv20190430(oc, oldCs) -> v20190430.ConvertFrom(oc, oldCs)

Collapse api packages to one directory:  github.com/openshift/openshift-azure/pkg/api/2019-04-30/api -> `github.com/openshift/openshift-azure/pkg/api/2019-04-30`
```

Admin api and pluginConfig is using stable api now.

### CODE REVIEW

**Comment by jim-minter** [comment]
/hold
looks encouraging, want to read it through carefully

**Comment by mjudeikis** [comment]
> /hold
> looks encouraging, want to read it through carefully

np, will do one more run myself tomorrow too. Would like to merge your PR first.

File: pkg/api/converterfromv20180930preview.go [hunk 1]
```diff
@@ -67,7 +67,7 @@ func mergeFromResourcePurchasePlanv20180930preview(oc *v20180930preview.OpenShif
 	}
 }
 
-func mergeProperties(oc *v20180930preview.OpenShiftManagedCluster, cs *OpenShiftManagedCluster) error {
+func mergePropertiesv20180930preview(oc *v20180930preview.OpenShiftManagedCluster, cs *OpenShiftManagedCluster) error {
```
Comment (jim-minter) [inline pkg/api/converterfromv20180930preview.go]: this is all a mouthful.
1) I can't remember why the converter code has to live in pkg/api - is it in any way feasible to move it out in to the subpackages?
2) failing that, create a `type v20180930converter struct{}` and make these methods of it - also less awful

**Comment by mjudeikis** [comment]
I want to remove all the references to names now and make it just:
```
import pluginapiconverter "github.com/openshift/openshift-azure/pkg/api/plugin/converter"
...
pluginapiconverter.From(template, &g.cs.Config, g.cs.Config.PluginVersion)
pluginapiconverter.To(template, &g.cs.Config, g.cs.Config.PluginVersion)
```

As basically all converters convert from "external api to Internal type and back

**Comment by mjudeikis** [comment]
* conclusion is to condence api and converters to one version folder
* keep converters splitted 
* and: 
```
pkg/api/2018-09-30-preview/types.go
package v20180930preview
```

**Comment by mjudeikis** [comment]
@jim-minter, ready for review. 
We have `test/utils/populate.Walk` and `util/structs.Walk` which shares a lot of code. I think they could be collapsed but I don't want to do it here. Followup?

**Comment by mjudeikis** [comment]
/retest
flake https://github.com/openshift/openshift-azure/issues/1362

**Comment by mjudeikis** [comment]
/retest 
same issue

**Comment by jim-minter** [comment]
@mjudeikis  this has come out very nicely.  I've added 4 commits at https://github.com/jim-minter/openshift-azure/commit/d62cc05e1c5be99d8471b46338f96f4db17017a0 - please can you see if you want them as well?

**Comment by jim-minter** [comment]
you know what, let's merge this, I'll open a 2nd PR for the others and you can take a look

**Comment by jim-minter** [comment]
> We have test/utils/populate.Walk and util/structs.Walk which shares a lot of code. I think they could be collapsed but I don't want to do it here. Followup?

@mjudeikis yes, post-GA - please open a cleanup issue

**Comment by mjudeikis** [comment]
> you know what, let's merge this, I'll open a 2nd PR for the others and you can take a look

Lets merge with this one. One less effort to merge + deadlines.

**Comment by mjudeikis** [comment]
upgrade failed with: https://github.com/openshift/openshift-azure/issues/1362

**Pull request merged**